src/sdk/service/iam-service.ts

```diff
@@ -151,4 +157,97 @@ export class IamService {
+            // are required for a custom policy and that it doesn't have any
+            // permissions that are not available to a custom policy
+            for (const domainPermission of availablePermissions) {
+              if (domainPermission.requiredForCustomPolicy && !policy.hasPermission(domainPermission.permissionType)) {
+                errors.push((new Error('Custom policy doesn\'t have the required permission.')));
+              } else if (!domainPermission.availableToCustomPolicy &&
+                policy.hasPermission(domainPermission.permissionType)) {
+                errors.push((new Error('Custom policy is not allowed to have the specified permission.')));
+              }
+            }
+          }
+        }
+        if (entityOptional.parentUuid !== null) {
+          const parentEntity = companyInventory.getEntityByUuid(entityOptional.parentUuid);
+          if (parentEntity) {
+            const effectiveParentPolicy = this.getEffectivePolicy(companyInventory, parentEntity, roleCreationRequest);
+            if (effectiveParentPolicy) {
+              let perm: Permission | undefined;
+              let impliedPerm: Permission | undefined;
+              for (const parentPermission of effectiveParentPolicy.permissions) {
+                perm = PermissionService.getPermission(parentPermission);
+                if (perm && perm.impliedPermissions) {
+                  for (const impliedPermission of perm.impliedPermissions) {
+                    impliedPerm = PermissionService.getPermission(impliedPermission);
+                    if (impliedPerm && impliedPerm.domain === policy.entityDomain &&
+                      !policy.hasPermission(impliedPermission)) {
+                      // the policy must have this permission, otherwise it violates an
+                      // implication of higher-level policies
+                      errors.push((new Error('Policy must contain permission=' + impliedPermission +
+                        ' since it is implied by a higher level permission.')));
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+        // verify that the policies has all permissions that are implied by its
+        // own permissions on the same domain
+        for (const permission of policy.permissions) {
+          const perm = PermissionService.getPermission(permission);
+          if (perm && perm.impliedPermissions) {
+            for (const impliedPermission of perm.impliedPermissions) {
+              const implPerm = PermissionService.getPermission(impliedPermission);
+              if (implPerm && implPerm.domain === policy.entityDomain && !policy.hasPermission(impliedPermission)) {
+                // the policy must have this permission, otherwise it violates an
+                // implication of higher-level policies
+                errors.push((new Error('Policy must contain permission=' + impliedPermission +
```

Comment:
again, need to include name of entity for the policy so that the message is useful to a user